src/main/java/edu/usf/cutr/gtfs_realtime/bullrunner/GtfsRealtimeProviderImpl.java

```diff
@@ -16,688 +16,446 @@
+            connection = URL_Request.openConnection();
+        } catch (Exception ex) {
+            _log.error("Error in opening feeds url", ex);
+        }
+        connection.setConnectTimeout(_Conn_Timeout_MS);
+        connection.setReadTimeout(_Conn_Timeout_MS);
+        InputStream in = connection.getInputStream();
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(in, "UTF-8"));
+
+        StringBuilder builder = new StringBuilder();
+        String inputLine;
+        JSONArray response;
+        try {
+            while ((inputLine = reader.readLine()) != null)
+                builder.append(inputLine).append("\n");
+            response = new JSONArray(builder.toString());
+        } catch (SocketTimeoutException ex) {
+            _log.error("Error readline, server doesn't close the connection.", ex);
+            response = null;
+        }
+
+        if (response.length() == 0)
+            return null;
+
+        // keep only vehicles of the given route_id
+        JSONArray response_out = new JSONArray();
+        for (int i = 0; i < response.length(); i++){
+            JSONObject vehicleObj = response.getJSONObject(i);
+            String vehicle_id = vehicleObj.getString("id");
+            // check pattern name
+            URL URL_pattern = new URL(_url + "v1/vehicles/" + vehicle_id + "/arrivals?count=100&api-key=" + _api_key);
+            try {
+                connection = URL_pattern.openConnection();
+            } catch (Exception ex) {
+                _log.error("Error in opening feeds url", ex);
+            }
+            connection.setConnectTimeout(_Conn_Timeout_MS);
+            connection.setReadTimeout(_Conn_Timeout_MS);
+            in = connection.getInputStream();
+
+            BufferedReader reader1 = new BufferedReader(new InputStreamReader(in, "UTF-8"));
+
+            StringBuilder builder1 = new StringBuilder();
+            String inputLine1;
+            JSONArray response_pattern;
+            try {
+                while ((inputLine1 = reader1.readLine()) != null)
+                    builder1.append(inputLine1).append("\n");
+                response_pattern = new JSONArray(builder1.toString());
+                String pattern_name = response_pattern.getJSONObject(0).getJSONObject("pattern").getString("name");
+            } catch (SocketTimeoutException ex) {
+                _log.error("Error readline, server doesn't close the connection.", ex);
+                response_pattern = null;
+                return null;
+            }
+            String pattern_name = response_pattern.getJSONObject(0).getJSONObject("pattern").getString("name");
+
+            // check if pattern name matches
+            if (pattern_name.equals(Pattern_name_C)){
+                if (route_id.equals(Route_id_C))
+                    response_out.put(vehicleObj);
+            } else if (pattern_name.equals(Pattern_name_MSC)){
+                if (route_id.equals(Route_id_MSC))

```

Comment:
@minhhpham I think `if (route_id.equals(Route_id_MSC))` is always true?  I'm going to remove it in my next commit, but please let me know if you think it needs to stay for some reason.

Reply:
Actually, looking more the value gets passed in from the JSON response, so on 2nd thought we do need it.  Let me know if I'm wrong.

Reply:
We need this to assign vehicles to the correct `route_id`. `refreshTripVehicle()` loops through `route_id` in GTFS files and request vehicle info from the API. Since the API returns both route C and MSC Express vehicle as a response to the request for route C, these 4 lines attempt to match `pattern_name` with `route_id`.